src/git-version.cr

```diff
@@ -68,7 +68,9 @@ module GitVersion
       branchTags.each do |tag|
         begin
           currentTag = SemanticVersion.parse(tag)
-          if (latestVersion < currentTag)
+          if (cb == MASTER_BRANCH && !currentTag.prerelease.to_s.empty?)
```

Comment:
I would remove the `cb == MASTER_BRANCH && ` check and the `to_s` should be not needed.

Reply:
It makes sense to compare versions with pre-releases in non master branches.
`2.0.1-SNAPSHOT.${hash} -> 2.0.2-SNAPSHOT.${hash}`

Reply:
The to_s is needed because the "prerelease" is not a string